synpurge/pg.py

```diff
@@ -67,6 +67,54 @@ def get_room_info(self, room_id):
     def find_table_indexes(self, table_name):
         return self._db.synapse.table_indexes(table_name)
 
+    def find_applications(self):
+        return self._db.synapse.application_services()
+
+    def find_rooms_by_application(self, application_id):
+        return self._db.synapse.rooms_by_application_service_id(application_id)
+
+    def delete_tuples_from_table_by_id(self, room_id):
+        tables_to_prune = ['current_state_events', 'event_json', 'event_auth',
+                           'event_backward_extremities', 'event_edges',
+                           'event_forward_extremities', 'event_push_actions',
+                           'event_search', 'event_reports', 'guest_access',
+                           'local_invites', 'topics', 'public_room_list_stream',
+                           'pusher_throttle', 'receipts_graph',
+                           'receipts_linearized', 'room_aliases',
+                           'room_account_data', 'room_depth', 'room_hosts',
+                           'room_memberships', 'room_names', 'room_tags',
+                           'room_tags_revisions', 'rooms',
+                           'stream_ordering_to_exterm', 'history_visibility',
+                           'feedback', 'state_events',
+                           'state_forward_extremities', 'state_groups_state',
+                           'state_groups', 'events']
+        for table in tables_to_prune:
```

Comment:
Probably this needs to go inside a transaction, otherwise I highly suspect that PostgreSQL will complain about failed checks on constraints. 

Using transactions with `py-postgresql` is done using `.xact()` ([docs here](http://pythonhosted.org/py-postgresql/driver.html#transactions)), which returns a context manager. That means you can do:

```python
with self._db.xact():
    for table in tables_to_prune:
        # Do the deletions on the database
```